data_collection/processing/event_type_detection.rst

```diff
@@ -0,0 +1,150 @@
+
+Outlink
+~~~~~~~~~~~~~~~~~~
+
+Event will be categorized as `Outlink` when `link` parameter of the tracked event is provided.
+
+Consent form impression
+~~~~~~~~~~~~~~~~~~
+
+Event will be categorized as `ConsentFormImpression` when `e_c` parameter of the tracked event has value `consent_form_impression` assigned.
+
+Consent form click
+~~~~~~~~~~~~~~~~~~
+
+Event will be categorized as `ConsentFormClick` when `e_c` parameter of the tracked event has value `consent_form_click` assigned.
+
+Consent decision
+~~~~~~~~~~~~~~~~~~
+
+Event will be categorized as `ConsentDecision` when `e_c` parameter of the tracked event has value `consent_decision` assigned.
+
+SharePoint
+~~~~~~~~~~~~~~~~~~
+SharePoint event type detection is a bit more complicated.
+For the event to be categorized as `SharePoint` type two things must occur:
+
+- Custom Variable 1 key has to be equal to `ppas.sharepoint.plugin`
+- `e_c` parameter has to be equal to either `download` or `search`
+
+Custom event
+~~~~~~~~~~~~~~~~~~
+
+Event will be categorized as `Custom` when `e_c` and `e_a` parameters of the tracked event are provided.
+
+Content interaction
+~~~~~~~~~~~~~~~~~~
+
+Event will be categorized as `ContentInteraction` when `c_i` and `c_n` parameters of the tracked event are provided.
+
+Content impression
+~~~~~~~~~~~~~~~~~~
+
+Event will be categorized as `ContentImpression` when only `c_n` parameter of the tracked event is provided (and `c_i` is not).
+
+Cart update
+~~~~~~~~~~~~~~~~~~
+
+Event will be categorized as `CartUpdated` when `idgoal` parameter of the tracked event is equal to `0` and `ec_id` parameter is NOT provided.
+
+Order completed
+~~~~~~~~~~~~~~~~~~
+
+Event will be categorized as `OrderCompleted` when `idgoal` parameter of the tracked event is equal to `0` but also `ec_id` parameter is provided.
+
+Site search
+~~~~~~~~~~~~~~~~~~
+
+Event will be categorized as `Search` when either `search`parameter of the tracked event is provided or a search term was detected in the tracked url (provided as the `url` parameter).
+
+Page view
+~~~~~~~~~~~~~~~~~~
+
+When every other detection step failed then your event will be categoried as a simple `PageView`.
+
+Special cases
+-----------------
+
+As you have probly noticed already, there are 3 event types missing in the detection process steps.
+
+- AbandonedCart
+- ExcludedEvent
+- BrokenEvent
+
+That is because those type are not "detected" but rather are a result of the post-processing of an event or a session.
+
+Abandoned Cart
+~~~~~~~~~~~~~~~~~~
+
+When a session did not track a `OrderCompleted` event, the last event of that type will be converted to `AbandonedCart`.
+
+Excluded Event
+~~~~~~~~~~~~~~~~~~
+
+There are several ways of excluding an event (e.g. by blacklisting source IP or User-Agent header matching).
+If an event matches given criteria it will be excluded from the reports but is still tracked and receives `ExcludedEvent` type.
```

Comment:
It is also worth to mention that ExcludedEvents are visible in Tracker Debugger.